pandas/core/groupby/generic.py

```diff
@@ -1181,6 +1197,63 @@ def blk_func(bvalues: ArrayLike) -> ArrayLike:
 
         return new_mgr
 
+    def _cython_agg_arrays(
+        self, how: str, alt=None, numeric_only: bool = True, min_count: int = -1
+    ) -> Tuple[List[ArrayLike], Index]:
+
+        data: ArrayManager = self._get_data_to_aggregate()
+
+        if numeric_only:
+            data = data.get_numeric_data(copy=False)
+
+        def py_fallback(bvalues: ArrayLike) -> ArrayLike:
+            # TODO(ArrayManager)
+            raise NotImplementedError
+
+        def array_func(values: ArrayLike) -> ArrayLike:
+
+            try:
+                result = self.grouper._cython_operation(
+                    "aggregate", values, how, axis=1, min_count=min_count
+                )
+            except NotImplementedError:
+                # generally if we have numeric_only=False
+                # and non-applicable functions
+                # try to python agg
+
+                if alt is None:
+                    # we cannot perform the operation
+                    # in an alternate way, exclude the block
+                    assert how == "ohlc"
+                    raise
+
+                result = py_fallback(values)
+
+            return _cast_agg_result(result, values, how, reshape1d=False)
+
+        result_arrays = [array_func(arr) for arr in data.arrays]
+
+        if not len(result_arrays):
+            raise DataError("No numeric types to aggregate")
+
+        return result_arrays, data.axes[0]
+
+    def _wrap_agged_arrays(self, arrays: List[ArrayLike], columns: Index) -> DataFrame:
+        if not self.as_index:
+            index = np.arange(arrays[0].shape[0])
+            mgr = ArrayManager(arrays, axes=[index, columns])
+            result = self.obj._constructor(mgr)
+            self._insert_inaxis_grouper_inplace(result)
+        else:
+            index = self.grouper.result_index
+            mgr = ArrayManager(arrays, axes=[index, columns])
+            result = self.obj._constructor(mgr)
```

Comment:
rather than adding new things & constructing the AM / BM here. can you do a pre-cursor PR that pushes this to an internals routine, maybe internals/blockwise. We do not want to keep adding code in here (and in fact want to remove the block manager / internals code).